src/hpc/tls.h

```diff
@@ -55,19 +55,40 @@ typedef struct ThreadLocalStorage
   UInt LocksContended;
 } ThreadLocalStorage;
 
-#ifdef HAVE_NATIVE_TLS
+#ifdef USE_NATIVE_TLS
 extern __thread ThreadLocalStorage *TLSInstance;
 #endif
 
+#ifdef USE_PTHREAD_TLS
+
+#include <pthread.h>
+
+#ifdef __GNUC__
+// This trick allows gcc/clang to hoist calls to pthread_getspecific()
+// out of loops and perform constant subexpression elimination on
+// expressions containing TLS values.
+__attribute__((pure)) void * pthread_getspecific(pthread_key_t);
+#endif
+
+static pthread_key_t TLSKeyCopy;
+pthread_key_t GetTLSKey(void);
+
+__attribute__((constructor))
+static void InitTLSKey() {
```

Comment:
Please clang-format

Reply:
Done. I'll probably also add checks for `__attribute__((constructor))` and `__attribute__((pure))` to the configure script rather than relying on ifdefs.

Reply:
You should be able to use AX_GCC_FUNC_ATTRIBUTE which we already use for other attributes.